src/Plugin/TfaSetup/TfaTotpSetup.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Drupal\ga_login\Plugin\TfaSetup;
+
+use Base32\Base32;
+use Drupal\Core\Link;
+use Drupal\tfa\Plugin\TfaSetupInterface;
+use Drupal\tfa\Plugin\TfaValidation\TfaTotp;
+use Drupal\Core\Url;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\user\UserDataInterface;
+use Drupal\user\Entity\User;
+
+/**
+ * @TfaSetup(
+ *   id = "tfa_totp_setup",
+ *   label = @Translation("TFA Totp Setup"),
+ *   description = @Translation("TFA Totp Setup Plugin"),
+ *   help_links = {
+ *    "Google Authenticator (Android/iPhone/BlackBerry)" = "https://support.google.com/accounts/answer/1066447?hl=en",
+ *    "Authy (Android/iPhone)" = "https://www.authy.com/thefuture#install-now",
+ *    "FreeOTP (Android)" = "https://play.google.com/store/apps/details?id=org.fedorahosted.freeotp",
+ *    "GAuth Authenticator (desktop)" = "https://github.com/gbraad/html5-google-authenticator"
+ *   }
+ * )
+ */
+class TfaTotpSetup extends TfaTotp implements TfaSetupInterface {
```

Comment:
I'm not sure, but if TFA can or should provide a base class with some/all of this code, the GA classes TfaTotpSetup and TfaHotpSetup would extend that class and provide ONLY it's overrides